### PR TITLE
Embeddings: minimal end-to-end searching with qdrant

### PR DESCRIPTION
**Pull request by camdencheek**
This is the final PR in the chain that makes end-to-end searching work locally with qdrant. All the changes in this PR are protected behind the `qdrant` feature flag, which should only be enabled locally because we do not yet have a qdrant instance running in any of our deployed instances.

## Test plan

Tested searching locally. PR is low risk because it is not actually active in production yet.

### CODE REVIEW

File: internal/codycontext/context.go [hunk 1]
```diff
@@ -308,6 +326,48 @@ func (c *CodyContextClient) getKeywordContext(ctx context.Context, args GetConte
...
 
+func (c *CodyContextClient) getEmbeddingsContextFromQdrant(ctx context.Context, args GetContextArgs) (_ []FileChunkContext, err error) {
+	embeddingsConf := conf.GetEmbeddingsConfig(conf.Get().SiteConfig())
+	if c == nil {
+		return nil, errors.New("embeddings not configured or disabled")
+	}
+	client, err := embed.NewEmbeddingsClient(embeddingsConf)
+	if err != nil {
+		return nil, errors.Wrap(err, "getting embeddings client")
+	}
```
Comment (keegancsmith) [inline internal/codycontext/context.go:337]: does creating a client here have a cost like dialing/etc? If so likely something to remember to follow-up on before we make this rolled out fully.

File: internal/codycontext/context.go [hunk 2]
```diff
@@ -308,6 +326,48 @@ func (c *CodyContextClient) getKeywordContext(ctx context.Context, args GetConte
...
+
+	resp, err := client.GetQueryEmbedding(ctx, args.Query)
+	if err != nil || len(resp.Failed) > 0 {
+		return nil, errors.Wrap(err, "getting query embedding")
+	}
+	query := resp.Embeddings
+
+	params := vdb.SearchParams{
+		ModelID:   client.GetModelIdentifier(),
+		RepoIDs:   args.RepoIDs(),
```
Comment (keegancsmith) [inline internal/codycontext/context.go:347]: out of interest what happens when this is empty? Currently the RepoIDs field is not documented.

File: internal/codycontext/context.go (hunk 1 above)
Comment (camdencheek) [reply internal/codycontext/context.go:337]: No, creating a new client here is very cheap right now. Dialing is lazy (using `httpcli.ExternalClient`). I created the client here to match the pattern that exists in the embeddings service, where we also create a client per request. Longer-term, I'd like to make the client a dependency of `CodyContextClient` so we can more easily mock out external requests.

File: internal/codycontext/context.go (hunk 2 above)
Comment (camdencheek) [reply internal/codycontext/context.go:347]: Oh, good catch. Right now, an empty set of repo IDs will not filter by repo (returning all results). In this case, that's not desirable. I just pushed a commit to protect call to `Search()` with an empty-repo-set check, and documented that RepoIDs will do no filtering if it's empty. I think we want to retain the option to provide no filtering, so I kept the check outside the `Search()` method.

**Pull request merged**